### PR TITLE
Add support to deploy a bundle

### PR DESCRIPTION
**Pull request by mcbenjemaa**
## Checklist

- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: api/capo/v1alpha4/group.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//+kubebuilder:object:generate=true
```
Comment (QuentinBisson) [inline api/capo/v1alpha4/group.go:1]: Why did you add this?

File: service/controller/resource/app/resource.go [hunk 2]
```diff
@@ -39,8 +39,7 @@ type Resource struct {
 // New creates a new configured app state getter resource managing
 // app CRs.
 //
-//     https://pkg.go.dev/github.com/giantswarm/resource/v2/appresource#StateGetter
-//
+//	https://pkg.go.dev/github.com/giantswarm/resource/v2/appresource#StateGetter
```
Comment (QuentinBisson) [inline service/controller/resource/app/resource.go:42]: Can you revert this ?

**Comment by mcbenjemaa** [comment]
@QuentinBisson Changes related to `InCluster` and `targetNamespaces` are adopted

**Review by uvegla — APPROVED** [review]

**Comment by uvegla** [comment]
Could you please add a few test cases to [create_test.go](https://github.com/giantswarm/cluster-apps-operator/blob/63cad22786ca9802afdcd1913d34238270dc010c/service/controller/resource/app/create_test.go) to cover the new condition?

**Comment by QuentinBisson** [comment]
I can next week, but I'm wondering if this change even make sense as the cluster-apps-operator only installs the chart and app operator if I understood correctly so it will never install neither defaults apps  nor bundles 🤔

**Comment by QuentinBisson** [comment]
Closing for now as it looks uneeded
